src/pyhf/constraints.py

```diff
@@ -219,10 +214,12 @@ def _precompute(self):
 
     def _dataprojection(self, auxdata):
         tensorlib, _ = get_backend()
-        auxdata = tensorlib.astensor(auxdata)
         poisson_data = tensorlib.gather(auxdata, self.poisson_data)
         return poisson_data
 
+    def has_pdf(self):
```

Comment:
Needs a docstring, but I can add this after we're done with all the benchmark testing so that I don't trigger unnecessarily long builds.

Reply:
done